doc/v5.0/man3/MPI_Type_get_extent.3
.\" -*- nroff -*-
.\" Copyright 2013 Los Alamos National Security, LLC. All rights reserved.
.\" Copyright 2010 Cisco Systems, Inc.  All rights reserved.
.\" Copyright 2006-2008 Sun Microsystems, Inc.
.\" Copyright (c) 1996 Thinking Machines Corporation
.\" Copyright (c) 2020      Google, LLC. All rights reserved.
.\" $COPYRIGHT$
.TH MPI_Type_get_extent 3 "Sep 30, 2021" "5.0.0rc1" "Open MPI"
.SH NAME
\fBMPI_Type_get_extent\fP, \fBMPI_Type_get_extent_x\fP \- Returns the lower bound and extent of a data type.

.SH SYNTAX
.ft R
.SH C Syntax
.nf
#include <mpi.h>
int MPI_Type_get_extent(MPI_Datatype \fIdatatype\fP, MPI_Aint\fI *lb\fP,
	MPI_Aint *\fIextent\fP)
int MPI_Type_get_extent_x(MPI_Datatype \fIdatatype\fP, MPI_Count\fI *lb\fP,
	MPI_Count *\fIextent\fP)

.fi
.SH Fortran Syntax (see FORTRAN 77 NOTES)
.nf
USE MPI
! or the older form: INCLUDE 'mpif.h'
MPI_TYPE_GET_EXTENT(\fIDATATYPE, LB, EXTENT, IERROR\fP)
	INTEGER	\fIDATATYPE, IERROR\fP
	INTEGER(KIND=MPI_ADDRESS_KIND) \fILB, EXTENT\fP
MPI_TYPE_GET_EXTENT_X(\fIDATATYPE, LB, EXTENT, IERROR\fP)
	INTEGER	\fIDATATYPE, IERROR\fP
	INTEGER(KIND=MPI_COUNT_KIND) \fILB, EXTENT\fP

.fi
.SH Fortran 2008 Syntax
.nf
USE mpi_f08
MPI_Type_get_extent(\fIdatatype\fP, \fIlb\fP, \fIextent\fP, \fIierror\fP)
	TYPE(MPI_Datatype), INTENT(IN) :: \fIdatatype\fP
	INTEGER(KIND=MPI_ADDRESS_KIND), INTENT(OUT) :: \fIlb\fP, \fIextent\fP
	INTEGER, OPTIONAL, INTENT(OUT) :: \fIierror\fP
MPI_Type_get_extent_x(\fIdatatype\fP, \fIlb\fP, \fIextent\fP, \fIierror\fP)
	TYPE(MPI_Datatype), INTENT(IN) :: \fIdatatype\fP
	INTEGER(KIND = MPI_COUNT_KIND), INTENT(OUT) :: \fIlb\fP, \fIextent\fP
	INTEGER, OPTIONAL, INTENT(OUT) :: \fIierror\fP

.fi
.SH INPUT PARAMETER
.ft R
.TP 1i
datatype
Data type (handle).
.sp
.SH OUTPUT PARAMETERS
.ft R
.TP 1i
lb
Lower bound of data type (integer).
.TP 1i
extent
Data type extent (integer).
.ft R
.TP 1i
IERROR
Fortran only: Error status (integer).

.SH DESCRIPTION
.ft R
MPI_Type_get_extent returns the lower bound and the extent of \fIdatatype\fP. For either function, if either the \fIlb\fP or \fIextent\fP parameter cannot express the value to be returned (e.g., if the parameter is too small to hold the output value), it is set to MPI_UNDEFINED.

.SH NOTE
.ft R
Use of MPI_Type_get_extent is strongly recommended over the old MPI-1 functions MPI_Type_extent and MPI_Type_lb.

.SH FORTRAN 77 NOTES
.ft R
The MPI standard prescribes portable Fortran syntax for
the \fILB\fP and \fIEXTENT\fP arguments only for Fortran 90. FORTRAN 77
users may use the non-portable syntax
.sp
MPI_Type_get_extent:
.sp
.nf
     INTEGER*MPI_ADDRESS_KIND \fILB\fP
or
     INTEGER*MPI_ADDRESS_KIND \fIEXTENT\fP
.fi
.sp
MPI_Type_get_extent_x:
.sp
.nf
     INTEGER*MPI_COUNT_KIND \fILB\fP
or
     INTEGER*MPI_COUNT_KIND \fIEXTENT\fP
.fi
.sp
where MPI_ADDRESS_KIND and MPI_COUNT_KIND are constants defined in mpif.h
and give the length of the declared integer in bytes.

.SH ERRORS
Almost all MPI routines return an error value; C routines as the value of the function and Fortran routines in the last argument.
.sp
Before the error value is returned, the current MPI error handler is
called. By default, this error handler aborts the MPI job, except for I/O function errors. The error handler may be changed with MPI_Comm_set_errhandler; the predefined error handler MPI_ERRORS_RETURN may be used to cause error values to be returned. Note that MPI does not guarantee that an MPI program can continue past an error.


doc/v1.4/man3/MPI_Type_get_extent.3.php
<?php
$topdir = "../../..";
$title = "MPI_Type_get_extent(3) man page (version 1.4.5)";
$meta_desc = "Open MPI v1.4.5 man page: MPI_TYPE_GET_EXTENT(3)";

include_once("$topdir/doc/nav.inc");
include_once("$topdir/includes/header.inc");
?>
<p> <a href="../">&laquo; Return to documentation listing</a></p>
  <!-- manual page source format generated by PolyglotMan v3.2, -->
<!-- available at http://polyglotman.sourceforge.net/ -->

<body bgcolor='white'>
<a href='#toc'>Table of Contents</a><p>

<h2><a name='sect0' href='#toc0'>Name</a></h2>
<b>MPI_Type_get_extent</b> - Returns the lower bound and extent of a data
type.
<p>
<h2><a name='sect1' href='#toc1'>Syntax</a></h2>

<h2><a name='sect2' href='#toc2'>C Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
int MPI_Type_get_extent(MPI_Datatype datatype, MPI_Aint *lb,
<tt> </tt>&nbsp;<tt> </tt>&nbsp;MPI_Aint *extent)
</pre>
<h2><a name='sect3' href='#toc3'>Fortran Syntax (see FORTRAN 77 NOTES)</a></h2>
<br>
<pre>INCLUDE &rsquo;mpif.h&rsquo;
MPI_TYPE_GET_EXTENT(DATATYPE, LB, EXTENT, IERROR)
<tt> </tt>&nbsp;<tt> </tt>&nbsp;INTEGER<tt> </tt>&nbsp;<tt> </tt>&nbsp;DATATYPE, IERROR
<tt> </tt>&nbsp;<tt> </tt>&nbsp;INTEGER(KIND=MPI_ADDRESS_KIND) LB, EXTENT
</pre>
<h2><a name='sect4' href='#toc4'>C++ Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
void MPI::Datatype::Get_extent(MPI::Aint&amp; lb, MPI::Aint&amp; extent)
<tt> </tt>&nbsp;<tt> </tt>&nbsp;const
</pre>
<h2><a name='sect5' href='#toc5'>Input Parameter</a></h2>

<dl>

<dt>datatype       </dt>
<dd>Data type (handle). <p>
</dd>
</dl>

<h2><a name='sect6' href='#toc6'>Output Parameters</a></h2>

<dl>

<dt>lb </dt>
<dd>Lower
bound of data type (integer). </dd>

<dt>extent       </dt>
<dd>Data type extent (integer). </dd>

<dt>IERROR
</dt>
<dd>Fortran only: Error status (integer).
<p> </dd>
</dl>

<h2><a name='sect7' href='#toc7'>Description</a></h2>
MPI_Type_get_extent returns
the lower bound and the extent of <i>datatype</i>.
<p>
<h2><a name='sect8' href='#toc8'>Note</a></h2>
Use of MPI_Type_get_extent
is strongly recommended over the old MPI-1 functions <a href="../man3/MPI_Type_extent.3.php">MPI_Type_extent</a> and
<a href="../man3/MPI_Type_lb.3.php">MPI_Type_lb</a>.
<p>
<h2><a name='sect9' href='#toc9'>Fortran 77 Notes</a></h2>
The MPI standard prescribes portable Fortran
syntax for the <i>LB</i> and <i>EXTENT</i> arguments only for Fortran 90. FORTRAN 77 users
may use the non-portable syntax <p>
<br>
<pre>     INTEGER*MPI_ADDRESS_KIND LB
or
     INTEGER*MPI_ADDRESS_KIND EXTENT
</pre><p>
where MPI_ADDRESS_KIND is a constant defined in mpif.h and gives the length
of the declared integer in bytes.
<p>
<h2><a name='sect10' href='#toc10'>Errors</a></h2>
Almost all MPI routines return
an error value; C routines as the value of the function and Fortran routines
in the last argument. C++ functions do not return errors. If the default
error handler is set to MPI::ERRORS_THROW_EXCEPTIONS, then on error the
C++ exception mechanism will be used to throw an MPI:Exception object. <p>
Before
the error value is returned, the current MPI error handler is called. By
default, this error handler aborts the MPI job, except for I/O function
errors. The error handler may be changed with <a href="../man3/MPI_Comm_set_errhandler.3.php">MPI_Comm_set_errhandler</a>; the
predefined error handler MPI_ERRORS_RETURN may be used to cause error values
to be returned. Note that MPI does not guarantee that an MPI program can
continue past an error.
<p>
<p> <p>

<hr><p>
<a name='toc'><b>Table of Contents</b></a><p>
<ul>
<li><a name='toc0' href='#sect0'>Name</a></li>
<li><a name='toc1' href='#sect1'>Syntax</a></li>
<li><a name='toc2' href='#sect2'>C Syntax</a></li>
<li><a name='toc3' href='#sect3'>Fortran Syntax (see FORTRAN 77 NOTES)</a></li>
<li><a name='toc4' href='#sect4'>C++ Syntax</a></li>
<li><a name='toc5' href='#sect5'>Input Parameter</a></li>
<li><a name='toc6' href='#sect6'>Output Parameters</a></li>
<li><a name='toc7' href='#sect7'>Description</a></li>
<li><a name='toc8' href='#sect8'>Note</a></li>
<li><a name='toc9' href='#sect9'>Fortran 77 Notes</a></li>
<li><a name='toc10' href='#sect10'>Errors</a></li>
</ul>


<p> <a href="../">&laquo; Return to documentation listing</a></p>
<?php
include_once("$topdir/includes/footer.inc");
